src/Middleware/RetryHandler.cs

```diff
@@ -248,5 +230,37 @@ private static bool ShouldRetry(HttpStatusCode statusCode)
                 _ => false
             };
         }
+
+        private static async Task<Exception> GetInnerException(HttpResponseMessage response, CancellationToken cancellationToken)
+        {
+            var httpStatusCode = response.StatusCode;
+            string? errorMessage = null;
+
+            // Drain response content to free connections. Need to perform this
+            // before retry attempt and before the TooManyRetries ServiceException.
+            if(response.Content != null)
+            {
+#if NET5_0_OR_GREATER
+                var responseContent = await response.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(false);
+#else
+                var responseContent = await response.Content.ReadAsByteArrayAsync().ConfigureAwait(false);

```

Comment:
```suggestion
                errorMessage = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
#else
                errorMessage = await response.Content.ReadAsStringAsync().ConfigureAwait(false);

```
This way we can get rid of the `GetStringFromContent` function.